tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -0,0 +1,288 @@
+   "source": [
+    "# copied from mode_purpose_share.ipynb\n",
+    "# We select participants from real data to be users at this time\n",
+    "participant_uuid_obj = list(edb.get_profile_db().find({\"install_group\": \"participant\"}, {\"user_id\": 1, \"_id\": 0}))\n",
+    "all_users = [u[\"user_id\"] for u in participant_uuid_obj]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adaptive-supplement",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logger = logging.getLogger()\n",
+    "logger.setLevel(logging.DEBUG)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "hollow-prefix",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logging.debug('There are %s users, we will plot the graphs for one of them' % len(all_users))\n",
+    "\n",
+    "user = all_users[1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "prospective-conducting",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.cluster_pipeline as eamtc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "finnish-istanbul",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# we should experiment with different values here\n",
+    "radius = 300"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "frequent-slovakia",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.similarity as similarity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bronze-switch",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.cluster_pipeline as pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dependent-machinery",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "trips = pipeline.read_data(uuid=user)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "round-death",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "bin_trips, bins = pipeline.remove_noise(trips, radius)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "latest-cooling",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "bin_start_lat = []\n",
+    "bin_start_log = []\n",
+    "\n",
+    "# Flip indices because points are in geojson (i.e. lon, lat)\n",
+    "for trip in bin_trips:\n",
+    "    trip_start_lat = trip.data.start_loc[\"coordinates\"][1]\n",
+    "    trip_start_log = trip.data.start_loc[\"coordinates\"][0]\n",
+    "    bin_start_lat.append(trip_start_lat)\n",
+    "    bin_start_log.append(trip_start_log)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "international-unemployment",
```

Comment:
what is the difference between `international-unemployment` and `latest-cooling` cells?

Reply:
I don't know this. It is neither the code or output. Is it important?

Reply:
I'm trying to point out that you have two cells that seem to have the same code. The `"id"` is not really important, just to show you the cells that seem to be the same.